ivadomed/losses.py

```diff
@@ -233,3 +234,60 @@ def forward(self, input, target):
 
         # TODO: Take the opposite?
         return focal_tversky_sum / n_classes
+
+
+class L2loss(nn.Module):
+
+    def __init__(self):
+        super(L2_loss, self).__init__()
+
+    def forward(self, input, target):
+        return torch.sum((input - target) ** 2) / 2
+
+
+class AdapWingLoss(nn.Module):
+
+    def __init__(self):
+        super(AdapWingLoss, self).__init__()
+    """
+    adaptive Wing loss : https://arxiv.org/abs/1904.07399
+
+    """
+    def forward(self,input,target):
+        theta = 0.5
+        alpha = 2.1
+        w = 14
+        e = 1
+        A = w * (1 / (1 + torch.pow(theta / e, alpha - target))) * (alpha - target) * torch.pow(theta / e,
+                                                                                              alpha - target - 1) * (1 / e)
+        C = (theta * A - w * torch.log(1 + torch.pow(theta / e, alpha - target)))
+
+        batch_size = target.size()[0]
+        hm_num = target.size()[1]
+
+        mask = torch.zeros_like(target)
+        # W=10
+        kernel = cv2.getStructuringElement(cv2.MORPH_RECT, (2, 2))
```

Comment:
Would it be possible to use scipy instead of cv2?
eg https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.ndimage.morphology.generate_binary_structure.html#scipy.ndimage.morphology.generate_binary_structure

Reply:
yes probably I'll add that on the todo list !

Reply:
Done. It runs now i need to verify performance and all.